tests/transforms/test_control.py

```diff
@@ -309,3 +309,92 @@ def ansatz(weights, wires):
     tape = expand_tape(tape)
     assert len(tape.operations) == 10
     assert all(o.name in {"CNOT", "CRX", "Toffoli"} for o in tape.operations)
+
+
+@pytest.mark.parametrize("diff_method", ["backprop", "parameter-shift"])

```

Comment:
@antalszava this was deliberate to an extent, because I don't think it matters to test multiple gradient methods.

That is, backprop and parameter-shift are very different logical flows, so worth testing both (prior, backprop worked but parameter-shift didn't!)

But in this case, parameter-shift and finite diff are logically identical; the bugfix is on the gradient transform level

Reply:
I see! Yes, that makes sense. Would the main concern be increasing the testing time?

Just curious because although the gradient methods use the same pipeline, in theory, it could bring more coverage to test finite diff. The way the transforms are called is identical, but the way the two transforms work is different.

Reply:
> Just curious because although the gradient methods use the same pipeline, 

Oh I think it is even more specific than that; the logic here simply forces `qml.ctrl` to be decomposed into operations that _support_ gradients. This is required for the gradient transforms to work, but happens _before_ the gradient transforms are applied.

So the bug/bugfix is completely independent of the gradient transform at all, so I'm not sure it's worth testing multiple gradient transforms?

Reply:
Right! Frankly, my suggestion here was more so just aiming to get more options to be tested to be fail-safe. The way `qml.ctrl` works could change, but the test would likely stay (and should work for all diff methods). Having said that, I'm happy to go without it, no worries.

Reply:
Feel free to add it in! I think my main worry was that finite differences is not always accurate, so I felt weird adding it to an integration test that is checking the gradient value 😬

Reply:
Sure :+1: It seems to be incorrect for JAX.